airflow/jobs/triggerer_job_runner.py

```diff
@@ -608,6 +609,17 @@ async def run_trigger(self, trigger_id, trigger):
                 self.log.info("Trigger %s fired: %s", self.triggers[trigger_id]["name"], event)
                 self.triggers[trigger_id]["events"] += 1
                 self.events.append((trigger_id, event))
+        except asyncio.CancelledError as err:
+            trigger_timeout = trigger.task_instance.trigger_timeout
+            if trigger_timeout:
+                if not trigger_timeout.tzinfo:
+                    trigger_timeout = trigger_timeout.replace(tzinfo=timezone.utc)
```

Comment:
curious why the the tzinfo line is necessary.  it seems that it should already be UTC.  are you just being defensive or did you encounter an issue?

Reply:
Yep, I encounter the timezone issue after I change the fallback [here](https://github.com/apache/airflow/blob/6becb7031618867bc253aefc9e3e216629575d2d/airflow/jobs/scheduler_job_runner.py#L940) to 1.0 for testing.